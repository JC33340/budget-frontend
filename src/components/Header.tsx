import HeaderButton from './HeaderButton';
import { Link } from 'react-router-dom';
import HeaderDropdown from './HeaderDropdown';

const Header = () => {
    return (
        <header
            className="bg-blue rounded-xl p-4 py-4 flex items-center justify-between"
            data-testid="header-testid"
        >
            <Link to="/">
                <p className="text-white text-2xl font-semibold">Budgeter</p>
            </Link>
            <div className="hidden md:flex items-center gap-x-6">
                <HeaderButton href="/transactions" text="Transactions" />
                <button
                    className="text-white bg-dark-red p-2 rounded-md font-semibold cursor-pointer"
                    onClick={() => {
                        localStorage.clear();
                        return window.location.reload();
                    }}
                >
                    Logout
                </button>
            </div>
            <HeaderDropdown />
        </header>
    );
};

export default Header;
