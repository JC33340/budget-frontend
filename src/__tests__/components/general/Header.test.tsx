import Header from '../../../components/Header';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';

test('snapshot test', () => {
    render(
        <BrowserRouter>
            <Header />
        </BrowserRouter>
    );
    const header = screen.getByTestId('header-testid');
    expect(header).toMatchSnapshot();
});
