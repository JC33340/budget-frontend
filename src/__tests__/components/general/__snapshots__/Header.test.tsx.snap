// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<header
  class="bg-blue rounded-xl p-4 py-4 flex items-center justify-between"
  data-testid="header-testid"
>
  <a
    href="/"
  >
    <p
      class="text-white text-2xl font-semibold"
    >
      Budgeter
    </p>
  </a>
  <div
    class="hidden md:flex items-center gap-x-6"
  >
    <a
      class="text-lg font-semibold p-2 rounded-md hover:bg-white hover:text-blue transition-all ease-in-out  text-white"
      data-testid="headerButton-testid"
      href="/transactions"
    >
      Transactions
    </a>
    <button
      class="text-white bg-dark-red p-2 rounded-md font-semibold cursor-pointer"
    >
      Logout
    </button>
  </div>
  <div
    class="flex md:hidden relative"
    data-testid="headerDropdown-testid"
  >
    <svg
      class="fill-white cursor-pointer"
      data-testid="dropdownButton-testid"
      fill="currentColor"
      height="40"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="40"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M432 176H80c-8.8 0-16-7.2-16-16s7.2-16 16-16h352c8.8 0 16 7.2 16 16s-7.2 16-16 16zM432 272H80c-8.8 0-16-7.2-16-16s7.2-16 16-16h352c8.8 0 16 7.2 16 16s-7.2 16-16 16zM432 368H80c-8.8 0-16-7.2-16-16s7.2-16 16-16h352c8.8 0 16 7.2 16 16s-7.2 16-16 16z"
      />
    </svg>
    <div
      class="absolute hidden flex-col right-0 bottom-[-190px] rounded-lg border-4 overflow-hidden border-dark-gray"
      data-testid="dropdownContent-testid"
    >
      <a
        class="cursor-pointer border-b-2 bg-white p-2 text-lg text-blue font-semibold hover:bg-blue hover:text-white transition-colors ease-in-out"
        href="/transactions"
      >
        Transactions
      </a>
      <a
        class="cursor-pointer border-b-2 bg-white p-2 text-lg text-blue font-semibold hover:bg-blue hover:text-white transition-colors ease-in-out"
        href="/reports"
      >
        Reports
      </a>
      <a
        class="cursor-pointer border-b-2 bg-white p-2 text-lg text-blue font-semibold hover:bg-blue hover:text-white transition-colors ease-in-out"
        href="/settings"
      >
        Settings
      </a>
      <button
        class="cursor-pointer bg-dark-red p-2 text-lg text-white font-semibold"
      >
        Log out
      </button>
    </div>
  </div>
</header>
`;
